sql/planner/projection.go

```diff
@@ -0,0 +1,209 @@
+package planner
+
+import (
+	"errors"
+
+	"github.com/genjidb/genji/database"
+	"github.com/genjidb/genji/document"
+	"github.com/genjidb/genji/sql/query"
+	"github.com/genjidb/genji/sql/query/expr"
+)
+
+// A ProjectionNode is a node that uses the given expressions to create a new document
+// for each document of the stream. Each expression can extract fields from the incoming
+// document, call functions, execute arithmetic operations. etc.
+type ProjectionNode struct {
+	node
+
+	Expressions []ResultField
+	tableName   string
+
+	cfg *database.TableConfig
+	tx  *database.Transaction
+}
+
+var _ outputNode = (*ProjectionNode)(nil)
+var _ operationNode = (*ProjectionNode)(nil)
+
+// NewProjectionNode creates a node that uses the given expressions to create a new document
+// for each document of the stream. Each expression can extract fields from the incoming
+// document, call functions, execute arithmetic operations. etc.
```

Comment:
I think the following is enough. It was redundant with the `ProjectionNode` documentation.
```suggestion
// NewProjectionNode creates a ProjectionNode.
```